src/include/storage/index/hash_index_builder.h

```diff
@@ -173,10 +176,20 @@ class PrimaryKeyIndexBuilder {
     common::PhysicalTypeID keyTypeID() const { return keyDataTypeID; }
 
 private:
-    template<typename T, typename S = T>
-    inline HashIndexBuilder<T, S>* getTypedHashIndex(uint64_t indexPos) {
-        return common::ku_dynamic_cast<InMemHashIndex*, HashIndexBuilder<T, S>*>(
-            hashIndexBuilders[indexPos].get());
+    template<common::IndexHashable T>
+    using HashIndexType =
+        typename std::conditional<std::same_as<T, std::string_view> || std::same_as<T, std::string>,
+            common::ku_string_t, T>::type;
+    template<typename T>
+    inline HashIndexBuilder<HashIndexType<T>>* getTypedHashIndex(uint64_t indexPos) {
```

Comment:
I wonder if you can simplify the `if constexpr` here by making use of `HashIndexType`.
maybe something like following
```cpp
    template<common::IndexHashable T>
    using HashIndexType = typename std::conditional<std::same_as<T, std::string_view> || std::same_as<T, common::ku_string_t> || std::same_as<T, std::string>, common::ku_string_t, T>::type;
    template<typename T>
    inline HashIndexBuilder<HashIndexType<T>>* getTypedHashIndex(uint64_t indexPos) {
        if constexpr (std::same_as<HashIndexType<T>, common::ku_string_t>) {
            return common::ku_dynamic_cast<InMemHashIndex*, HashIndexBuilder<common::ku_string_t>*>(
                hashIndexBuilders[indexPos].get());
        } else {
            return common::ku_dynamic_cast<InMemHashIndex*, HashIndexBuilder<T>*>(
                hashIndexBuilders[indexPos].get());
        }
    }
```